backend/app/models/evaluation_task.py

```diff
@@ -0,0 +1,64 @@
+import enum
+from uuid import UUID
+from typing import Optional, List
+
+from sqlalchemy import Text, JSON
+
+from sqlmodel import (
+    Field,
+    Column,
+    String,
+    Relationship as SQLRelationship,
+)
+
+from app.models.base import UpdatableBaseModel
+
+
+class EvaluationStatus(str, enum.Enum):
+    NOT_START = "not_start"
+    EVALUATING = "evaluating"
+    DONE = "done"
+    ERROR = "error"
+
+
+class EvaluationTask(UpdatableBaseModel, table=True):
+    id: Optional[int] = Field(default=None, primary_key=True)
+    name: str = Field(max_length=255)
+
+    user_id: UUID = Field(foreign_key="users.id", nullable=True)
+    user: "User" = SQLRelationship(
+        sa_relationship_kwargs={
+            "lazy": "joined",
+            "primaryjoin": "EvaluationTask.user_id == User.id",
+        },
+    )
+
+    dataset_id: int = Field(nullable=True)
+
+    evaluation_task_items: List["EvaluationTaskItem"] = SQLRelationship(back_populates="evaluation_task")
+
+    __tablename__ = "evaluation_tasks"
+
+
+class EvaluationTaskItem(UpdatableBaseModel, table=True):
+    id: Optional[int] = Field(default=None, primary_key=True)
+    chat_engine: str = Field(max_length=255)
+    status: EvaluationStatus = Field(sa_column=Column(String(32), nullable=False))
+    query: str = Field(sa_column=Column(Text))
+    reference: str = Field(sa_column=Column(Text))
+    response: str = Field(sa_column=Column(Text))
+    retrieved_contexts: list[str] = Field(default=[], sa_column=Column(JSON))
+    extra: dict = Field(default={}, sa_column=Column(JSON))
+    error_msg: str = Field(sa_column=Column(Text, nullable=True))
+    factual_correctness: Optional[float] = Field(nullable=True)
+    semantic_similarity: Optional[float] = Field(nullable=True)
```

Comment:
I have some concerns about the extensibility of this data model:
- How should we handle other evaluation metrics if we have more?
- How to handle user-defined custom metrics?

Although it may be a bit early to consider, table schema migration costs are relatively high.

langfuse's data model may be of some help with us:
https://langfuse.com/docs/scores/data-model

Reply:
It is too early to consider. If we need to add another metric, the effort will be way more than adding a field in the table. And before that, we should fix the retrieved_contexts first. For the user-defined custom metrics, we cannot support this version. PTAL of the `ragas.metrics` package.

Reply:
I perfered to store the metrics value in another table like:

```python
class EvaluationTaskItemScore(SQLModel, table=true):
   id: int
   name: varchar(40) # maybe `factual_correctness`, `semantic_similarity`, `faithfulness` and more ...
   value: float
   evaluation_task_item_id: int
   evaluation_task_id: int
```

Just a suggestion and reminder of extensibility, the hard code columns way is also ok for me.

cc:@wd0517 @sykp241095 What do you think?

Reply:
Langfuse supports user-defined metrics, so storing them in a separate table is required. However, we currently only support a limited set of hardcoded metrics, making a dedicated column for these acceptable for now.